src/PostgREST/Config.hs

```diff
@@ -165,77 +160,78 @@ instance JustIfMaybe a a where
-        <*> (fromMaybe False <$> optBool "db-channel-enabled")
-        <*> (maybe ["public"] splitOnCommas <$> optValue "db-extra-search-path")
-        <*> optWithAlias (optInt "db-max-rows")
-                         (optInt "max-rows")
-        <*> (fromMaybe 10 <$> optInt "db-pool")
-        <*> (fromMaybe 10 <$> optInt "db-pool-timeout")
-        <*> optWithAlias (optString "db-pre-request")
-                         (optString "pre-request")
-        <*> (fromMaybe True <$> optBool "db-prepared-statements")
-        <*> optWithAlias (optString "db-root-spec")
-                         (optString "root-spec")
-        <*> (fromList . splitOnCommas <$> reqWithAlias (optValue "db-schemas")
-                                                       (optValue "db-schema")
-                                                       "missing key: either db-schemas or db-schema must be set")
-        <*> (fromMaybe True <$> optBool "db-config")
-        <*> parseTxEnd "db-tx-end" snd
-        <*> parseTxEnd "db-tx-end" fst
-        <*> parseDbUri "db-uri"
-        <*> (fmap parseSecret <$> pSec)
-        <*> parseJwtAudience "jwt-aud"
-        <*> parseRoleClaimKey "jwt-role-claim-key" "role-claim-key"
-        <*> pSec
-        <*> pB64
-        <*> parseLogLevel "log-level"
-        <*> parseOpenAPIServerProxyURI "openapi-server-proxy-uri"
-        <*> (maybe [] (fmap encodeUtf8 . splitOnCommas) <$> optValue "raw-media-types")
-        <*> (fromMaybe "!4" <$> optString "server-host")
-        <*> (fromMaybe 3000 <$> optInt "server-port")
-        <*> (fmap T.unpack <$> optString "server-unix-socket")
-        <*> parseSocketFileMode "server-unix-socket-mode"
-
     parseDbUri :: C.Key -> C.Parser C.Config Text
-    parseDbUri k = flip fromMaybe dbUriFile <$> reqString k
+    parseDbUri k = reqString k
 
-    parseJwtSecret :: C.Key -> Bool -> C.Parser C.Config (Maybe B.ByteString)
-    parseJwtSecret k isB64 = optString k >>= \case
+    parseJwtSecret :: C.Key -> C.Parser C.Config (Maybe B.ByteString)
+    parseJwtSecret k = optString k >>= \case
       Nothing -> pure Nothing
-      Just sec  ->
-        let secStr = encodeUtf8 sec
-            secFile = fromMaybe secStr secretFile
-            -- replace because the JWT is actually base64url encoded which must be turned into just base64 before decoding.
-            replaceUrlChars = T.replace "_" "/" . T.replace "-" "+" . T.replace "." "="
-            willBeFile = isPrefixOf "@" (toS secStr) && isNothing secretFile
-        in
-        if isB64 && not willBeFile -- don't decode in bas64 if the secret will be a file or it will err. The secFile will be filled with the file contents in a later stage.
-          then case B64.decode . encodeUtf8 . T.strip . replaceUrlChars $ decodeUtf8 secFile of
-            Left errMsg -> fail errMsg
-            Right bs    -> pure $ Just bs
-          else pure $ Just secFile
+      Just s  -> pure . Just $ encodeUtf8 s
```

Comment:
Those two can probably be (simplified and) moved just in the list above, similar to all the others?